tcms/testplans/templates/testplans/mutable.html

```diff
@@ -92,8 +92,8 @@
             <div class="form-group">
                 <label class="col-md-12 col-lg-12 {% if form.text.errors %}has-error{% endif %}" for="id_document"> {% trans "Test plan document:" %}</label>
                 <div class="col-lg-12 col-md-12">
-                    <div>{{ form.text }}</div>
-                    {{ form.text.errors }}
+                    <input id="simplemde-file-upload" type="file" style="display: none">
+                    <textarea type="text" name="text" id="text" class="form-control"></textarea>
```

Comment:
Instead of removing core/widgets.py and repeating this block into every existing template that uses it why not leave the templates intact and just call `initSimpleMDE()` inside SimpleMDE.render() with the necessary default arguments ?  Looks much cleaner to me and less related changes.

Reply:
I haven't though of that, but even when you mention it now I don't think it's such a good idea. Here's why - putting the call to `initSimpleMDE` inside a python widget would introduce an implicit dependency to a js function that can be refactored/renamed, etc. I was already confused when I refactored this, because the js file was containing a variable called `simplemde` that was coming from nowhere, and in the end it turned out it was coming from the python file. 

I think it's much better to have a clear separation between the python views/widgets and js code. For the reasons I mentioned above.

Also, even if we did put this into a python widget we would still need to duplicate the media imports for example, in the comments page where we have more than 1 SimpleMDE form. 

All in all, these Django features are great if you want to write less code, and are OK, because they just work... until they don't and you have refactor all of them (like now).

If you want to reduce the duplication, I suggest we extract this snipet into a template and the include it like:
```
{% include "simple_mde_editor.html" with id="text" and file_input_id="simplemde-file-upload" %}
```
then we can even put the `initSimpleMDE` call in the same template and not repeat that.